### DISCUSSION TITLE
🐬 GeoVista v0.4.0

### DISCUSSION THREAD

**Discussion by bjlittle** [bjlittle/geovista#393] [Releases]
<div align="center">

| Minor Release | Status | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v0.4.0` | [![status](https://img.shields.io/badge/v0.4.0-released-green)](https://github.com/bjlittle/geovista/discussions/393) | `18-09-2023` | N/A | [v0.4.0](https://github.com/bjlittle/geovista/releases/tag/v0.4.0) | [![conda-forge](https://img.shields.io/badge/v0.4.0-orange?label=conda-forge&logo=conda-forge&logoColor=white)](https://anaconda.org/conda-forge/geovista?version=0.4.0&channel=main) | [![pypi](https://img.shields.io/pypi/v/geovista/0.4.0?color=orange&label=pypi&logo=python&logoColor=white)](https://pypi.org/project/geovista/0.4.0) |

| Patch Release | Status | Date | RTD | Release | conda-forge | PyPI |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| `v0.4.1` | [![status](https://img.shields.io/badge/v0.4.1-released-green)](https://github.com/bjlittle/geovista/discussions/393) | `10-01-2024` | N/A | [v0.4.1](https://github.com/bjlittle/geovista/releases/tag/v0.4.1) | [![conda-forge](https://img.shields.io/badge/v0.4.1-orange?label=conda-forge&logo=conda-forge&logoColor=white)](https://anaconda.org/conda-forge/geovista?version=0.4.1&channel=main) | [![pypi](https://img.shields.io/pypi/v/geovista/0.4.1?color=orange&label=pypi&logo=python&logoColor=white)](https://pypi.org/project/geovista/0.4.1) |

</div>

**Comment by bjlittle** [comment]
I'm keen to bank some additional new functionality for v0.4.0 including the following, all of which are currently under development: 

- [x] `geovista.transform.transform_point` (https://github.com/bjlittle/geovista/pull/409)
- [x] `geovista.transform.transform_points` (https://github.com/bjlittle/geovista/pull/409)
- [x] `geovista.geoplotter.GeoPlotterBase.add_points` (https://github.com/bjlittle/geovista/pull/442)

The release strategy at the moment is simply to be light and nimble. The current `geovista` user base is small, so releasing little and often is the primary focus so that we can get rapid-ish mileage and feedback on the growing codebase.

**Comment by bjlittle** [comment]
The 0.4.0 release is targeting outcomes from https://github.com/bjlittle/geovista/labels/scipy%202023, which was an amazing conference 🥳

**Comment by bjlittle** [comment]
I'll add a wee bit more rigour to future releases in an attempt to sign-post the features that we're aiming to deliver i.e., a release roadmap.

Hopefully this will whet the appetite of the community and give a heads-up on what's happening and when 👍

**Comment by bjlittle** [comment]
Apologies, this release slipped slightly to the right after some well overdue down-time after scipy.

I'm aiming to release 0.4.0 within the next 2 weeks 👍

**Reply by dennissergeev** [reply]
I'm being a bit cheeky, but is there any chance to slip a simple wind vector example into this release please? Specifically, a script like the one introduced in #387 but for cubed-sphere (LFRic) data. I tried it myself but there's something wrong with the data length and I didn't have enough knowledge & time to dig into that...

<details>

```python
mesh.arrows
```

gives me error:
```python
ValueError: data length of (13824) != required length (13826)
```

</details>

**Reply by bjlittle** [reply]
Thanks for the heads-up ... I'll see if I can squeeze something in after I nail of the functionality that I need to bank. But if we fail to manage that, we can work on maturing wind vector support as a priority along with volume rendering for 0.5.0.

Does that sound reasonable?

**Reply by bjlittle** [reply]
Also, can you easily share you're code with me so that I can see what's happening ?

**Reply by dennissergeev** [reply]
Here's a quick-and-dirty script:

https://gist.github.com/dennissergeev/4cb8a4773f6d08c88ce4241dfe568236

The data file in netCDF is attached, but I'll also email it to you just in case.

**Reply by bjlittle** [reply]
@dennissergeev Sorry, clearly this was a tad late to make the cut for `0.4.0`.

Do you fancy making an issue?

I'm starting to think and plan `0.5.0` 🤔 

I'll make an associated `0.5.0` GH Project to track deliverables, and we can add your issue to make sure this gets addressed and banked 👍

**Reply by dennissergeev** [reply]
Thanks @bjlittle! Should we just re-animate issue #16 for this? Or would you like me to open a fresh one?

**Reply by bjlittle** [reply]
Yeah, let's pull in #16, that's a great call 💯

**Comment by bjlittle** [comment]
The `v0.4.1` patch release is now available!

This removes the `pyvista` pin, allowing us to leverage all the latest features :partying_face: 

We now have a min pin of `pyvista>=0.43.1`

**Reply by tkoyama010** [reply]
Awesome job!

**Reply by bjlittle** [reply]
Thanks for all your help to make it happen! 

Really appreciated 🍻
